backend/src/permissions/permissions.service.ts

```diff
@@ -38,6 +40,43 @@ export class PermissionsService {
     private noteConfig: NoteConfig,
     private eventEmitter: EventEmitter2<NoteEventMap>,
   ) {}
+  /**
+   * Checks if the given {@link User} is has the in {@link desiredPermission} specified permission on {@link Note}.
+   *
+   * @async
+   * @param {Permission} desiredPermission - permission level to check for
+   * @param {User} user - The user whose permission should be checked. Value is null if guest access should be checked
+   * @param {Note} note - The note for which the permission should be checked
+   * @return if the user has the specified permission on the note
+   */
+  public async checkPermissionOnNote(
+    desiredPermission: Permission,
```

Comment:
```suggestion
    desiredPermission: Exclude<Permission, Permission.CREATE>,
```
See https://www.typescriptlang.org/docs/handbook/utility-types.html#excludeuniontype-excludedmembers

Reply:
changed